pkg/controller/controller_suite_test.go

```diff
@@ -555,6 +555,25 @@ func waitForCacheSync(stop <-chan struct{}, controller *controller) {
 	)).To(BeTrue())
 }
 
+func EqualTaints(a, b []corev1.Taint) bool {
```

Comment:
Not sure, if it makes sense to have such function here, open for suggestions.
@ggaurav10 - this is from our conversation yesterday.

Reply:
lgtm. Given that array of taints is not ordered, `reflect.DeepEqual()` cannot be used.
Just as a suggestion: WDYT about generic `assert.Equal()`.
https://github.com/stretchr/testify#assert-package

The PR also works though.

Reply:
Ah yes, I think it could be used, but wondering if it has a dependency on the `testing` package, not sure if we can use otherwise.

Reply:
`testing` package is already depended on by `ginkgo` and `gomega`.

Reply:
But since `gomega` is already being used, please use [`ConsistOf`](https://onsi.github.io/gomega/#consistofelement-interface).

Reply:
Perfect, I missed this. I can use this in all occurrences where taints are being compared. 
thanks for the suggestion.

Reply:
Here, we want both slices to be exactly equal[ignoring order], so we should check the length separately right?

Reply:
How about [`MatchYAML`](https://onsi.github.io/gomega/#matchyamlyaml-interface)?

Reply:
I just made changes with ConsistOf.  :D

Reply:
> Here, we want both slices to be exactly equal[ignoring order], so we should check the length separately right? or using a separate method could be a better option. , is there a provided-matcher for that as well.

Length check is [built into](https://github.com/onsi/gomega/blob/master/matchers/consist_of.go#L42) `ConsistOf`. So additional length check is not required.